### DISCUSSION TITLE
Example for speech enhancement

### DISCUSSION THREAD

**Discussion by JNaranjo-Alcazar** [asteroid-team/asteroid#449] [Q&A]
Hi community! I was wondered if there is an example for speech enhancement. That is, given a noisy audio, the framework will return a cleaned audio with the speech.

Thanks in advance!

**Comment by mpariente** [comment]
Here is a demo, built by @JorisCos 
https://asteroid-demo.herokuapp.com/

BTW, thanks for opening in discussion, this is the right place! :smiley:

**Reply by JNaranjo-Alcazar** [reply]
Thanks for the info!
Is the code used available somewhere? I mean, I can see an example about how to do a source separation, but I cannot see a speech enhancement one...
Thanks again!

**Comment by mpariente** [comment]
https://github.com/asteroid-team/asteroid_app

**Reply by JNaranjo-Alcazar** [reply]
Thank you very much partner! I definitely owe you one ;)

I'm still starting out in the speech domain... It's quite different from environmental audio haha

**Comment by mpariente** [comment]
By the way, you can also use the inference capabilities of the HF hub models, [here](https://huggingface.co/mhu-coder/ConvTasNet_Libri1Mix_enhsingle) for example. 
We didn't enable it for all of them, but it should be simple to do.

**Reply by JNaranjo-Alcazar** [reply]
Ok, I'll have a look at the models that are available :)

**Comment by manu0586** [comment]
I just tried some pretrained models for SE  
Basically you can perform speech enhancement in 2 lines (Big thanks at ASTEROID TEAM for that ) :

```
from asteroid.models import BaseModel
#SE-Model
if ID=='DPT':
    model= BaseModel.from_pretrained("JorisCos/DPTNet_Libri1Mix_enhsingle_16k")  #si-SDR : 14.7
if ID=='DPRNN':
    model=BaseModel.from_pretrained("JorisCos/DPRNNTasNet-ks2_Libri1Mix_enhsingle_16k") #si-SDR : 14.7
if ID=='DCCRN':
    model=BaseModel.from_pretrained("JorisCos/DCCRNet_Libri1Mix_enhsingle_16k")
if ID=='ConvTasNet':
    model=BaseModel.from_pretrained("JorisCos/ConvTasNet_Libri1Mix_enhsingle_16k") 

model.separate('test.wav')
```
The enhanced recording is saved in 'test_est1.wav'

**Comment by jonashaag** [comment]
You can also use the CLI to do inference, see the basic tutorial: http://colab.research.google.com/github/asteroid-team/asteroid/blob/master/notebooks/00_GettingStarted.ipynb

The tutorial is for separation but works the same for enhancement. Please tell if there is anything we could have explained better!
